pkg/synthetictests/events.go

```diff
@@ -72,19 +79,45 @@ func testDuplicatedEvents(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
 	const testName = "[sig-arch] events should not repeat pathologically"
 
 	displayToCount := map[string]int{}
+	ignoredCounts := map[string]int{}
 	for _, event := range events {
 		eventDisplayMessage, times := getTimesAnEventHappened(fmt.Sprintf("%s - %s", event.Locator, event.Message))
-		if times > 20 {
-			if knownEvents.MatchString(eventDisplayMessage) {
-				continue
-			}
+		if times == 0 {
+			continue
+		}
+
+		prev, ok := displayToCount[eventDisplayMessage]
+		// If we see the event for the first time, but it has already happened
+		// multiple, usually it means that previous events happened before the
+		// tests run. There might be exceptions (for example, combined events)
+		// if the event message changes.
+		if !ok && times > 1 && !strings.Contains(eventDisplayMessage, "(combined from similar events): ") {
+			displayToCount["before tests: "+eventDisplayMessage] = times - 1
```

Comment:
I don't agree with this change.  The fact that there's an ongoing issue is a problem.

Reply:
It's not an ongoing issue. Let's say you have a pod with 15 readiness probe failures during installation, and it has 7 failures due to reconfiguration that is made by a test. In total it has 22 failures, but they happened at complete different times and have different reasons. An event counter has a flaw that these events are counted together.